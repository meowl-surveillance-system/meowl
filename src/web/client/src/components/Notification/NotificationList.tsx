import React, { Component } from "react";
import { Container, Typography } from "@material-ui/core";
import Notification from "./Notification";

interface Props {}
interface State {
  list: Array<Record<string, any>>;
}

interface Notif {
  date: Date;
  type: string;
  name: string;
  frame_id: string;
  stream_id: string;
}

/* 
 * A component for rendering multiple notifications. Each individual notification is rendered
 * using the Notification component
 */
export default class NotificationList extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      list: [],
    };
  }

  componentDidMount() {
    fetch(`/notif/retrieveNotifications`)
      .then((res) => res.json())
      .then((result) => {
        result.map((element: Notif) => {
          const tmp = {
            date: new Date(),
            type: "",
            name: "",
            img: "",
          };
          fetch(`/notif/retrieveFrame/${element.frame_id}/${element.stream_id}`)
            .then((res) => res.json())
            .then((result) => {
              tmp.date = element.date;
              tmp.type = element.type;
              tmp.name = element.name;
              tmp.img = result.frame.data;
              this.setState({ list: [...this.state.list, tmp] });
            })
            .catch((e) => console.error(e));
        });
      })
      .catch((e) => console.error(e));
  }

  /*
   * Maps each element in list state to a Notification component based on that element's
   * date, type, name, and frame buffer
   */
  render() {
    return (
      <Container>
        <Typography>Notifications</Typography>
        {this.state.list.map((notif: any, index: number) => (
          <Notification
            key={index}
            date={notif.date}
            type={notif.type}
            name={notif.name}
            img={notif.img}
          />
        ))}
      </Container>
    );
  }
}
