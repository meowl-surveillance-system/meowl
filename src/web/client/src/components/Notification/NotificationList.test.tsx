import React from "react";
import { render } from "@testing-library/react";
import NotificationList from "./NotificationList";

describe('NotificationList component', () => {
  it('renders NotificationList component', () => {
    const { container } = render(
      <NotificationList />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
})