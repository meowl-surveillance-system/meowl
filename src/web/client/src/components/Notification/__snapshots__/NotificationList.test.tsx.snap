// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationList component renders NotificationList component 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg"
>
  <p
    class="MuiTypography-root MuiTypography-body1"
  >
    Notifications
  </p>
</div>
`;
